data/creaturescripts/scripts/boss.lua

```diff
@@ -0,0 +1,226 @@
+	local argc = #argv
+	for k, v in ipairs(argv) do
+		table.insert(buffer, v)
+		if k < argc and sep then
+			table.insert(buffer, sep)
+		end
+	end
+end
+
+local function insertItems(buffer, info, parent, items)
+	local start = info.running
+	for _, item in ipairs(items) do
+		if _ ~= 1 or parent > 100 then
+			table.insert(buffer, ",")
+		end
+		info.running = info.running + 1
+		table.insert(buffer, "(")        
+		pushSeparated(buffer, ",", info.playerGuid, parent, info.running, item:getId(), item:getSubType(), db.escapeString(serializeAttributes(item)))
+		table.insert(buffer, ")")
+
+		if item:isContainer() then
+			local size = item:getSize()
+			if size > 0 then              
+				local subItems = {}
+				for i = 1, size do
+					table.insert(subItems, item:getItem(i - 1))
+				end
+
+				insertItems(buffer, info, info.running, subItems)
+			end
+		end
+	end
+	return info.running - start
+end
+
+local function insertRewardItems(playerGuid, timestamp, itemList)
+	db.asyncStoreQuery('select `pid`, `sid` from `player_rewards` where player_id = ' .. playerGuid .. ' order by `sid` ASC;', 
+		function(query)
+			local lastReward = 0
+			local lastStoreId   
+			if(query) then             
+				repeat
+					local sid = result.getDataInt(query, 'sid')
+					local pid = result.getDataInt(query, 'pid')
+
+					if pid < 100 then
+						lastReward = pid
+					end
+					lastStoreId = sid
+				until not result.next(query)
+			end
+
+			local buffer = {'INSERT INTO `player_rewards` (`player_id`, `pid`, `sid`, `itemtype`, `count`, `attributes`) VALUES'}
+
+			--reward bag
+			local info = {
+				playerGuid = playerGuid,
+				running = lastStoreId or 100
+			}
+
+			local bag = Game.createItem(ITEM_REWARD_CONTAINER)
+			bag:setAttribute(ITEM_ATTRIBUTE_DATE, timestamp)
+			if itemList then
+				for _, p in ipairs(itemList) do
+					bag:addItem(p[1], p[2])
+				end
+			end
+
+			local total = insertItems(buffer, info, lastReward + 1, {bag})
+			table.insert(buffer, ";")
+
+			if total ~= 0 then
+				db.query(table.concat(buffer))
+			end
+		end
+	)
+end
+
+local function getPlayerStats(bossId, playerGuid, autocreate)
+	local ret = globalBosses[bossId][playerGuid]
+	if not ret and autocreate then
+		ret = {
+			bossId = bossId, 
+			damageIn = 0, -- damage taken from the boss
+			healing = 0, -- healing (other players) done
+		}
+		globalBosses[bossId][playerGuid] = ret
+		return ret
+	end
+	return ret
+end
+
+function onDeath(creature, corpse, killer, mostDamageKiller, lastHitUnjustified, mostDamageUnjustified)
+	local monsterType = creature:getType()
+	if monsterType:isRewardBoss() then -- Make sure it is a boss
+		local bossId = creature:getId()
+		local timestamp = os.time()
+
+		local totalDamageOut, totalDamageIn, totalHealing = 0.1, 0.1, 0.1 -- avoid dividing by zero
+
+		local scores = {}
+		local info = globalBosses[bossId]
+		local damageMap = creature:getDamageMap()
+
+		for guid, stats in pairs(info) do
+			local player = Player(stats.playerId)
```

Comment:
If the player gives damage to the boss and die before the boss dies, the server crashes because the player does not think the solution I found was to add in ondeath function:
	                if not player then
		        return true
	                end